internal/service/resources/collector/collector_acm.go

```diff
@@ -414,3 +449,110 @@ func (d *ACMDataSource) convertClusterToResourcePool(from data.Object) (to model
 
 	return
 }
+
+// makeCapacityInfo creates a map of strings of capacity attributes for the cluster
+func (d *ACMDataSource) makeCapacityInfo(cluster v1.ManagedCluster) map[string]string {
+	results := make(map[string]string)
+	tags := []string{"cpu", "ephemeral-storage", "hugepages-1Gi", "hugepages-2Mi", "memory", "pods"}
+	for _, tag := range tags {
+		if value, found := cluster.Status.Allocatable[v1.ResourceName(tag)]; found {
+			results[tag] = value.String()
+		}
+	}
+	return results
+}
+
+// getClusterClientConfig retrieves the cluster client config for the specified cluster.
+func (d *ACMDataSource) getClusterClientConfig(ctx context.Context, clusterName string) (*clientcmdapi.Config, error) {
+	// Fetch the kubeconfig for this cluster
+	kubeConfig, err := k8s.GetClusterKubeConfigFromSecret(ctx, d.hubClient, clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster kubeconfig: %w", err)
+	}
+
+	config, err := clientcmd.Load(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal cluster config: %w", err)
+	}
+
+	return config, nil
+}
+
+// getDeploymentManagerExtensions builds the extensions required for the deployment manager
+func (d *ACMDataSource) getDeploymentManagerExtensions(ctx context.Context, clusterName string) (map[string]string, error) {
+	// Fetch the kubeconfig for this cluster and provide the info as extensions to the returned object. This
+	// is anticipated as a temporary measure since eventually SMO will require accessing the API using an OAuth token
+	// acquired from an OAuth server.
+	config, err := d.getClusterClientConfig(ctx, clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster config: %w", err)
+	}
+
+	var caData, url string
+	if cluster, found := config.Clusters[clusterName]; found {
+		caData = string(cluster.CertificateAuthorityData)
+		url = cluster.Server
+	}
+
+	var adminUser, adminCert, adminKey string
+	if authInfo, found := config.AuthInfos["admin"]; found {
+		adminUser = "admin"
+		adminCert = string(authInfo.ClientCertificateData)
+		adminKey = string(authInfo.ClientKeyData)
+	}
+
+	return map[string]string{
+		"profileName":          "k8s",
+		"admin_client_cert":    adminCert,
+		"admin_client_key":     adminKey,
+		"admin_user":           adminUser,
+		"cluster_ca_cert":      caData,
+		"cluster_api_endpoint": url,
+	}, nil
+}
+
+// convertManagedClusterToDeploymentManager converts a ManagedCluster to a DeploymentManager object
```

Comment:
Please remind me the relationships again

DeploymentManager from ManagedCluster
ResourcePool also from ManagedCluster
Resource from Node which for SNO is only one
ResourceType for each Resource, meaning a Node

is that right?

Reply:
My point is, it seems you already have info enough to build the DeploymentManager, no?

Reply:
In this PR, yes that's correct.  That will get us parity with how the API currently works in the older implementation.  We need to change the handling of cluster so that it also has a resourceType so that we can raise alarms against it.   Need to get clarification from the standards folks on how they want that done.